packages/app/src/components/UnstakeForm.tsx

```diff
@@ -13,18 +13,28 @@ interface UnstakeFormProps {
   delayedWithdraw: boolean;
   setDelayedWithdraw: (delayedWithdraw: boolean) => void;
   gSolBalance: BN | undefined;
+  calculateFee?: (amount: BN) => Promise<BN>;
 }
 
 const UnstakeForm: React.FC<UnstakeFormProps> = ({
   withdraw,
   delayedWithdraw,
   setDelayedWithdraw,
+  calculateFee,
   gSolBalance,
 }) => {
   const [amount, setAmount] = useState("");
+  const [feeLoading, setFeeLoading] = useState(false);
 
   const withdrawModal = useModal(() => withdraw(amount));
 
+  const withdrawalFee = useMemo(() => {
+    if (!amount || !calculateFee) return new BN(0);
+
+    setFeeLoading(true);
+    return calculateFee(new BN(amount)).finally(() => setFeeLoading(false));
+  }, [calculateFee, amount]);
+
```

Comment:
Whenever I set the amount, I get the following error:
stakeAccount.ts:75 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'client')
    at calculateWithdrawalFee (stakeAccount.ts:75:1)
    at UnstakeForm.tsx:35:1
    at updateMemo (react-dom.development.js:17246:1)
    at Object.useMemo (react-dom.development.js:17886:1)
    at useMemo (react.development.js:1650:1)
    at UnstakeForm (UnstakeForm.tsx:31:1)
    at renderWithHooks (react-dom.development.js:16305:1)
    at updateFunctionComponent (react-dom.development.js:19588:1)
    at beginWork (react-dom.development.js:21601:1)
    at beginWork$1 (react-dom.development.js:27426:1)